### PR TITLE
le_tc/timer: Add corner cases and missing API's in timer TC

### PR DESCRIPTION
**Pull request by thapav**
This patch adds missing corner cases for following API's:
timer_create()
timer_delete()
timer_settime()
timer_gettime()
timer_getoverrun()

### CODE REVIEW

File: apps/examples/testcase/le_tc/kernel/tc_timer.c [hunk 1]
```diff
@@ -88,8 +110,50 @@ static void tc_timer_timer_create_delete(void)
...
+	st_sigevent.sigev_signo = sig_no;
+	st_sigevent.sigev_value.sival_ptr = &timer_id;
+
+	/* Check for NULL clock ID != CLOCK_REALTIME */
+
+	ret_chk = timer_create(clockid, &st_sigevent, &timer_id);
+	TC_ASSERT_NEQ("timer_create", ret_chk, ERROR);
+	TC_ASSERT_NEQ("timer_create", timer_id, NULL);
+
+	ret_chk = timer_getoverrun(&timer_id);
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/kernel/tc_timer.c]: the parameter of timer_getoverrun is timer_t type. is & needed?

File: apps/examples/testcase/le_tc/kernel/tc_timer.c [hunk 2]
```diff
@@ -122,11 +188,50 @@ static void tc_timer_timer_set_get_time(void)
...
+
+	/* Null it_value parameter check for timer_settime */
+
+	st_timer_spec_set.it_value.tv_sec = -1;
+	ret_chk = timer_settime(timer_id, 0, &st_timer_spec_set, NULL);
+	TC_ASSERT_EQ_CLEANUP("timer_settime", ret_chk, OK, timer_delete(timer_id));
+	st_timer_spec_set.it_value.tv_sec = 1;
+
+	/* Check if TIMER_ABSTIME (Flag = 1) is selected */
+	/* Check if TIMER_ABSTIME (Flag = 1) is selected */
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/kernel/tc_timer.c]: please remove unnecessary line

File: apps/examples/testcase/le_tc/kernel/tc_timer.c [hunk 3]
```diff
@@ -44,15 +44,37 @@ static void tc_timer_timer_create_delete(void)
 	int ret_chk = 0;
 	timer_t timer_id;
 	timer_t gtimer_id;
-	clockid_t clockid = CLOCK_REALTIME;
+	clockid_t clockid = CLOCK_MONOTONIC;
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/kernel/tc_timer.c]: CLOCK_MONOTONIC is defined in time.h and it has ifdef condition. Please check this.

**Comment by seinfra** [comment]
Code Rule Check Result:
apps/examples/testcase/le_tc/kernel/tc_timer.c:122: ERROR: [IDT_M_TAB] code indent should use tabs where possible

File: apps/examples/testcase/le_tc/kernel/tc_timer.c [hunk 4]
```diff
@@ -44,15 +44,37 @@ static void tc_timer_timer_create_delete(void)
...
 	struct posix_timer_s *st_ret_prt;
 	struct itimerspec st_timer_spec_val;
+
 	/* Set and enable alarm */
+
 	st_sigevent.sigev_notify = SIGEV_SIGNAL;
 	st_sigevent.sigev_signo = sig_no;
 	st_sigevent.sigev_value.sival_ptr = &timer_id;
 
+	/* Check for NULL clock ID != CLOCK_REALTIME */
```
Comment (lokeshbv) [inline apps/examples/testcase/le_tc/kernel/tc_timer.c]: check comments, Is it check for ` NULL clock ID ` or `clock ID != CLOCK_REALTIME `case ??

**Comment by seinfra** [comment]
Code Rule Check OK.

File: apps/examples/testcase/le_tc/kernel/tc_timer.c [hunk 5]
```diff
@@ -122,11 +193,48 @@ static void tc_timer_timer_set_get_time(void)
...
 
+	/* Null timer ID check for timer_settime */
+
+	ret_chk = timer_settime(NULL, 0, &st_timer_spec_set, NULL);
+	TC_ASSERT_EQ_CLEANUP("timer_settime", ret_chk, ERROR, timer_delete(timer_id));
+	TC_ASSERT_EQ_CLEANUP("timer_settime", errno, EINVAL, timer_delete(timer_id));
+
+	/* Null value check for timer_settime */
+
+	ret_chk = timer_settime(timer_id, 0, 0, NULL);
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/kernel/tc_timer.c]: 3rd param type is struct itimerspec *.
you'd better use NULL instead of 0

**Comment by jeongchanKim** [comment]
I think CI server fail is configuration problem.
After merging https://github.com/Samsung/TizenRT/pull/1117, it will be okay.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by thapav** [comment]
@jeongchanKim now it seems to work fine
Please review

File: apps/examples/testcase/le_tc/kernel/tc_timer.c (hunk 5 above)
Comment (thapav) [reply apps/examples/testcase/le_tc/kernel/tc_timer.c]: done

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by sunghan-chang** [comment]
@thapav Could you fill details on commit message what you add?

**Pull request merged**